lib/services/api/api_service.ts

```diff
@@ -172,6 +178,15 @@ export class SNApiService extends PureService {
     return response;
   }
 
+  async signOut() {
```

Comment:
What happens if you attempt to signOut with an expired token?

Reply:
This is an authenticated action, so the server will return a `498` status code but `snjs` will not perform the request to refresh the token.

Perhaps we should terminate the session from the server, even if the token is expire.

Reply:
I imagine this case will not be uncommon. A user returning to a device they haven't accessed in months will attempt to sign out to refresh stale data. So we do want a solution to this.

Reply:
> A user returning to a device they haven't accessed in months will attempt to sign out to refresh stale data.

I'm not following, why do they need to sign out in this case? If they haven't used the app for several month, only the `access_token` will be expired, but not the `refresh_token`. So if they open the app, the `refresh_token` will be used to extend the session for an additional year.

`access_token` will be valid for 60 days, but the `refresh_token` can be used for up to a year.

Reply:
Not saying that the user is rational ;) In many cases if something goes wrong in general, a user will attempt to sign out and sign back in. This is actually advice our support team will give to users in last-resort cases.

Reply:
Oh, I got you now :smile: 

> Not saying that the user is rational ;)

:laughing: